File: easyml/src/test/java/android/graphics/RectF.java
package android.graphics;

/**
 * Lightweight mock of android.graphics.RectF for local JVM unit testing.
 * Prevents unit test failures caused by unmocked Android stubs.
 */
public class RectF {
    public float left;
    public float top;
    public float right;
    public float bottom;

    public RectF() {}

    public RectF(float left, float top, float right, float bottom) {
        this.left = left;
        this.top = top;
        this.right = right;
        this.bottom = bottom;
    }

    public RectF(RectF r) {
        if (r != null) {
            this.left = r.left;
            this.top = r.top;
            this.right = r.right;
            this.bottom = r.bottom;
        }
    }

    public float width() {
        return right - left;
    }

    public float height() {
        return bottom - top;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        RectF rectF = (RectF) o;
        return Float.compare(rectF.left, left) == 0 &&
               Float.compare(rectF.top, top) == 0 &&
               Float.compare(rectF.right, right) == 0 &&
               Float.compare(rectF.bottom, bottom) == 0;
    }

    @Override
    public String toString() {
        return "RectF(" + left + ", " + top + ", " + right + ", " + bottom + ")";
    }
}

File: easyml/src/test/kotlin/android/graphics/RectF.kt
package android.graphics

/**
 * Lightweight mock of android.graphics.RectF for local JVM unit testing.
 * Prevents unit test failures caused by unmocked Android stubs.
 */
class RectF(
    @JvmField var left: Float = 0f,
    @JvmField var top: Float = 0f,
    @JvmField var right: Float = 0f,
    @JvmField var bottom: Float = 0f
) {
    constructor(r: RectF?) : this(
        left = r?.left ?: 0f,
        top = r?.top ?: 0f,
        right = r?.right ?: 0f,
        bottom = r?.bottom ?: 0f
    )

    fun width(): Float = right - left
    fun height(): Float = bottom - top

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is RectF) return false
        return left == other.left && top == other.top && right == other.right && bottom == other.bottom
    }

    override fun hashCode(): Int {
        var result = left.hashCode()
        result = 31 * result + top.hashCode()
        result = 31 * result + right.hashCode()
        result = 31 * result + bottom.hashCode()
        return result
    }

    override fun toString(): String = "RectF($left, $top, $right, $bottom)"
}
